iepy/preprocess/ner/regexp.py

```diff
@@ -0,0 +1,109 @@
+import re
+import codecs
+
+from nltk.text import TokenSearcher as NLTKTokenSearcher
+
+from iepy.preprocess.ner.base import BaseNERRunner
+
+
+class RegExpNERRunner(BaseNERRunner):
+
+    def __init__(self, label, regexp, override=False):
+        super(RegExpNERRunner, self).__init__(override=override)
+        self.label = label
+        self.regexp = regexp
+
+    def run_ner(self, doc):
+        entities = []
+        tokens = doc.tokens
+        searcher = TokenSearcher(tokens)
+        for match in searcher.finditer(self.regexp):
+            entity_oc = self.process_match(match)
+            if type(entity_oc) == list:
+                entities.extend(entity_oc)
+            else:
+                entities.append(entity_oc)
+        return entities
+
+    def process_match(self, match):
+        name = ' '.join(match.group())
+        kind = self.label
+        offset, offset_end = match.span()
+        entity_oc = self.build_occurrence(name, kind, name, offset, offset_end)
+
+        return entity_oc
+
+
+class TokenSearcher(NLTKTokenSearcher):
+    """
+    From nltk.text.TokenSearcher documentation:
+
+        A class that makes it easier to use regular expressions to search
+        over tokenized strings.  The tokenized string is converted to a
+        string where tokens are marked with angle brackets -- e.g.,
+        ``'<the><window><is><still><open>'``.  The regular expression
+        passed to the ``findall()`` method is modified to treat angle
+        brackets as non-capturing parentheses, in addition to matching the
+        token boundaries; and to have ``'.'`` not match the angle brackets.
+    """
+
+    def __init__(self, tokens):
+        # replace < and > inside tokens with \< and \>
+        _raw = '><'.join(w.replace('<', '\<').replace('>', '\>') for w in tokens)
+        # preprend >< instead of < for easier token counting
+        self._raw = '><' + _raw + '>'
+
+    def finditer(self, regexp):
+        regexp = preprocess_regexp(regexp)
+
+        i = re.finditer(regexp, self._raw)
+        while True:
+            try:
+                m = next(i)
+                start, end = m.span()
+                # FIXME: do not count from the beggining
```

Comment:
Sorry, my bad. This is not a FIXME, it is at most a TODO, because a small optimization can be done here. I think the comment can be safely removed.